pallets/asset-manager/src/lib.rs

```diff
@@ -0,0 +1,393 @@
+    }
+
+    /// Mapping from an asset id to asset type.
+    /// This is mostly used when receiving transaction specifying an asset directly,
+    /// like transferring an asset from this chain to another.
+    #[pallet::storage]
+    #[pallet::getter(fn asset_id_type)]
+    pub type AssetIdType<T: Config> = StorageMap<_, Blake2_128Concat, T::AssetId, T::AssetType>;
+
+    /// Reverse mapping of AssetIdType. Mapping from an asset type to an asset id.
+    /// This is mostly used when receiving a multilocation XCM message to retrieve
+    /// the corresponding asset in which tokens should me minted.
+    #[pallet::storage]
+    #[pallet::getter(fn asset_type_id)]
+    pub type AssetTypeId<T: Config> = StorageMap<_, Blake2_128Concat, T::AssetType, T::AssetId>;
+
+    /// Stores the units per second for local execution for a AssetType.
+    /// This is used to know how to charge for XCM execution in a particular
+    /// asset
+    /// Not all assets might contain units per second, hence the different storage
+    #[pallet::storage]
+    #[pallet::getter(fn asset_type_units_per_second)]
+    pub type AssetTypeUnitsPerSecond<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AssetType, u128>;
+
+    // Supported fee asset payments
+    #[pallet::storage]
+    #[pallet::getter(fn supported_fee_payment_assets)]
+    pub type SupportedFeePaymentAssets<T: Config> = StorageValue<_, Vec<T::AssetType>, ValueQuery>;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Register new asset with the asset manager
+        #[pallet::weight(T::WeightInfo::register_asset())]
+        pub fn register_asset(
+            origin: OriginFor<T>,
+            asset: T::AssetType,
+            metadata: T::AssetRegistrarMetadata,
+            min_amount: T::Balance,
+            is_sufficient: bool,
+        ) -> DispatchResult {
+            T::AssetModifierOrigin::ensure_origin(origin)?;
+
+            let asset_id: T::AssetId = asset.clone().into();
+            ensure!(
+                AssetIdType::<T>::get(&asset_id).is_none(),
+                Error::<T>::AssetAlreadyExists
+            );
+            T::AssetRegistrar::create_asset(asset_id, min_amount, metadata.clone(), is_sufficient)
+                .map_err(|_| Error::<T>::ErrorCreatingAsset)?;
+
+            AssetIdType::<T>::insert(&asset_id, &asset);
+            AssetTypeId::<T>::insert(&asset, &asset_id);
+
+            Self::deposit_event(Event::AssetRegistered {
+                asset_id,
+                asset,
+                metadata,
+            });
+            Ok(())
+        }
+
+        /// Change the amount of units we are charging per execution second for a given AssetType
+        #[pallet::weight(T::WeightInfo::set_asset_units_per_second(*num_assets_weight_hint))]
+        pub fn set_asset_units_per_second(
+            origin: OriginFor<T>,
+            asset_type: T::AssetType,
+            units_per_second: u128,
+            num_assets_weight_hint: u32,
```

Comment:
what is `num_assets_weight_hint` for?

Reply:
defined [here](https://github.com/rocketman1997/parallel-1/blob/9508db5cf074b53e4ae3d18618d6f2614a479533/pallets/asset-manager/src/lib.rs#L219) as to linearly increase weight/transaction_fee based on num of weighted/billable asset. yes, the name is a bit confusing maybe better as `WeightedAssetsNum`

Reply:
remove for simplicity